### DISCUSSION TITLE
Content blocker at the device level

### DISCUSSION THREAD

**Discussion by Slion** [Slion/Fulguris#351] [Ideas]
@Helium314 Have you ever considered NetGuard? There was some interesting discussions about that on our Discord channel.
https://github.com/M66B/NetGuard

It looks like it does content block at the device level instead of the application level.
If NetGuard does the job in theory we could remove our Fulguris implementation all together.
You could contribute to NetGuard if they are missing features, or fork it, if the team there does not want to integrate your features.
It's not that I don't want your changes in the Fulguris code base, don't read me wrong 😁 I'm very happy you contribute all that stuff.
I'm genuinely wondering if there are a better, more generic way to do that.
If that is indeed the way forward we could just redirect people to NetGuard when they try to enable the ad blocker in Fulguris, for instance.

**Comment by Helium314** [comment]
In general, blocking on device level, and having per-app settings (like NedGuard has) is definitely preferrable.

I only had a quick look, but
* NetGuard cannot be used together with other VPNs. Probably doesn't matter for most users
* The description and mention of hosts-files makes it look very much like DNS-based blocking, which is effectively very similar to the old blocker from Lightning browser. Without decrypting a https request, NetGuard can't even know which URL a browser is trying to access. (looks like [AdGuard](https://adguard.com/en/adguard-android/overview.html) is doing this)
